### PR TITLE
✨ Implements variable substitution macro for html element opacity.

### PR DESCRIPTION
**Pull request by joel-regus**
This change creates a new variable substitution macro called "opacity" that returns a selected html element's level of transparency, the values range from 0 to 1.

This change also takes into account the element's location in the DOM tree. It is possible for an element to have its opacity set to 1 (fully visible), yet its parent can be set to 0 (fully invisible). The goal is to return the opacity value that is visible by a human. If we are unable to return an element's opacity then the default value returned is 1.

### CODE REVIEW

**Comment by joel-regus** [comment]
I signed it!

File: src/inabox/inabox-viewport.js [hunk 1]
```diff
@@ -174,6 +174,27 @@ export class ViewportBindingInabox {
         });
   }
 
+  /** @override */
+  getElementOpacity(el) {
```
Comment (lannka) [inline src/inabox/inabox-viewport.js]: To not blow the size of AMP runtime, we'd like this to be moved to amp-analytics extension.

So no change needed to resource.js, viewport-*.js.
You can probably introduce a new file opacity.js under `amp-analytics/0.1/` and have all the logic there.

File: extensions/amp-analytics/0.1/opacity.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  return nodeList;
+
+}
+
+/**
+  *  Returns the min opacity found amongst the element and its parents
+  *  @param {!Element|null} el
+  *  @return {number} minimum opacity value
+  */
+export function getOpacity(el) {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/opacity.js]: nitpick: pls put public functions at the top

Comment (lannka) [inline extensions/amp-analytics/0.1/opacity.js]: should we multiply all the opacity values together?
it min opacity is really what you want, let's rename the method.

File: extensions/amp-analytics/0.1/opacity.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {computedStyle} from '../../../src/style';
+
+
+/**
+  *  Returns the min opacity found amongst the element and its parents
```
Comment (lannka) [inline extensions/amp-analytics/0.1/opacity.js]: nit: parents -> ancestors

**Comment by joel-regus** [comment]
@lannka thank you for the review! I have made the changes you requested.

**Comment by joel-regus** [comment]
@lannka should we make the PR to expose this as a macro or will that be handled on your end?

**Comment by lannka** [comment]
@joel-regus thinking twice, let's not worry about that, as it's already exposed in the `state` object.

**Comment by joel-regus** [comment]
@lannka is this ready to merge?

**Comment by nainar** [comment]
Pinging @lannka for next steps here.

**Comment by joel-regus** [comment]
@lannka is there any guidance you can provide in regards to these failing unit tests? We are not sure as to how to proceed.

**Pull request merged**